### PR TITLE
Added file transmission test

### PR DESCRIPTION
**Pull request by ethouris**

### CODE REVIEW

**Comment by ethouris** [comment]
NOTE: tests fail because of a bug that is fixed in #1236

File: test/test_file_transmission.cpp [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        size_t n = ifile.read(buf.data(), 1456).gcount();
+        size_t shift = 0;
+        while (n > 0)
+        {
+            int st = srt_send(sock_clr, buf.data()+shift, n);
+            ASSERT_GT(st, 0);
+
+            n -= st;
+            shift += st;
+            //std::cout << "(" << st << "+" << n << ")" << std::flush;
```
Comment (maxsharabayko) [inline test/test_file_transmission.cpp:146]:
```suggestion
```

File: test/test_file_transmission.cpp [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        }
+
+        if (ifile.eof())
+        {
+            //std::cout << "EOF\n(n=" << n << ")\n";
```
Comment (maxsharabayko) [inline test/test_file_transmission.cpp:151]:
```suggestion
```

File: test/test_file_transmission.cpp [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+#include <fstream>
+#include <ctime>
+#include <vector>
+
+//#pragma comment (lib, "ws2_32.lib")
+
+TEST(Transmission, FileUpload)
+{
+    srt_startup();
+    //srt_setloglevel(SRT_LOG_LEVEL_MAX);
```
Comment (maxsharabayko) [inline test/test_file_transmission.cpp:42]:
```suggestion
```

File: test/test_file_transmission.cpp [hunk 4]
```diff
@@ -0,0 +1,190 @@
+/*
+ * SRT - Secure, Reliable, Transport
+ * Copyright (c) 2018 Haivision Systems Inc.
```
Comment (maxsharabayko) [inline test/test_file_transmission.cpp:3]:
```suggestion
 * Copyright (c) 2020 Haivision Systems Inc.
```

File: test/test_file_transmission.cpp [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+ * 
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * 
+ * Submited by: Russell Greene (Issue #440)
```
Comment (maxsharabayko) [inline test/test_file_transmission.cpp:9]:
```suggestion
 * Based on the proposal by Russell Greene (Issue #440)
```

**Pull request merged**